plugin_commands.py

```diff
@@ -48,9 +60,52 @@ def on_result_async(self, edits: Union[JoinLinesRequest.ReturnType, Error], docu
         if document_version != self.view.change_count():
             return
         if isinstance(edits, Error):
-            print('[{}] Error handling the "{}" request. Falling back to native join.'.format(
-                self.session_name, JoinLinesRequest.Type))
+            sublime.status_message('Error handling the "{}" request. Falling back to native join.'.format(
+                JoinLinesRequest.Type))
             self.view.run_command('join_lines')
             return
         if edits:
             apply_text_edits_to_view(edits, self.view)
+
+
+class RustAnalyzerMoveItemCommand(RustAnalyzerCommand):
+
+    def run(self, edit: sublime.Edit, direction: Optional[MoveItemRequest.Direction] = None) -> None:
+        if direction not in ('Up', 'Down'):
+            sublime.status_message('Error running command: direction must be either "Up" or "Down".')
+            return
+        sublime.set_timeout_async(lambda: self.make_request_async(direction))
+
+    def make_request_async(self, direction: MoveItemRequest.Direction) -> None:
+        session = self.session_by_name(self.session_name)
+        if session is None:
+            return
+        session_view = session.session_view_for_view_async(self.view)
+        if not session_view:
+            return
+        view_listener = session_view.listener()
+        if not view_listener:
+            return
+        first_selection = first_selection_region(self.view)
+        if first_selection is None:
+            return
```

Comment:
There is potential for code reuse here since this has a lot of similarities with Join Lines command but I'd rather improve that in a dedicated PR.